### PR TITLE
Feature/boilerplate generation

### PR DESCRIPTION
**Pull request by mr-nick17**
Initial commit of new service created using dp tool

### CODE REVIEW

File: ci/scripts/audit.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -eux
+
+pushd dp-frontend-filter-flex-dataset
+  make audit
+popd
```
Comment (ChrysmOre) [inline ci/scripts/audit.sh:5]: Newline missing

File: handlers/handlers.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"net/http"
+
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/config"
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/mapper"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// ClientError is an interface that can be used to retrieve the status code if a client has errored
+type ClientError interface {
+	Error() string
```
Comment (ChrysmOre) [inline handlers/handlers.go:14]: It's unnecessary to include Error() in this interface, when you assert for retrieving the statusCode you're only interested in the Code() function.

File: handlers/handlers.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	Code() int
+}
+
+func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
+	status := http.StatusInternalServerError
+	if err, ok := err.(ClientError); ok {
+		if err.Code() == http.StatusNotFound {
```
Comment (ChrysmOre) [inline handlers/handlers.go:21]: This seems counter-productive. If you're going to allow retrieving the statusCode you shouldn't restrict it to one specific statusCode. I think this should be
```
if err, ok := err.(ClientError); ok {
  status = err.Code()
}
```

File: main.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	// Create service initialiser and an error channel for service errors
+	svcList := service.NewServiceList(&service.Init{})
+	svcErrors := make(chan error, 1)
+
+	// Read config
+	cfg, err := config.Get()
+	if err != nil {
+		log.Error(ctx, "unable to retrieve service configuration", err)
```
Comment (ChrysmOre) [inline main.go:38]: Logging errors and then also returning the same error? Please kill it with fire! We don't need errors being logged random amounts of times. Errors should be returned all the way up the call stack and logged once.
If you want to add some descriptive context (and you do) you should combine these 2 statements like this:
```
return fmt.Errorf("unable to retrieve service configuration: %w", err)
```
This returns the original error wrapped in the extra message and is a far preferable way of doing things.

File: service/service.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	svc.Config = cfg
+	svc.ServiceList = serviceList
+
+	// Initialise clients
+	clients := routes.Clients{}
+
+	// Get healthcheck with checkers
+	svc.HealthCheck, err = serviceList.GetHealthCheck(cfg, BuildTime, GitCommit, Version)
+	if err != nil {
+		log.Fatal(ctx, "failed to create health check", err)
```
Comment (ChrysmOre) [inline service/service.go:48]: Same here with these errors

File: service/service.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+		err := errors.New("failed to shutdown gracefully")
+		log.Error(ctx, "failed to shutdown gracefully ", err)
+		return err
+	}
+
+	log.Info(ctx, "graceful shutdown was successful")
+	return nil
+}
+
+func (svc *Service) registerCheckers(ctx context.Context, c routes.Clients) (err error) {
```
Comment (ChrysmOre) [inline service/service.go:124]: Don't use named return values if you don't need to. If you must, especially don't use named return values and then return explicit values, do one or the other.

**Review by ChrysmOre — COMMENTED** [review]
I know this code was auto-generated but there are a number of bad patterns and practices (as there are in all the auto-generated code) and IMO it would be better to fix now

**Review by dandwal — APPROVED** [review]

**Pull request merged**